emulator/remote_state_test.go

```diff
@@ -0,0 +1,92 @@
+package emulator
+
+import (
+	"context"
+	"encoding/hex"
+	"testing"
+
+	"github.com/onflow/flow-go/fvm/evm/emulator/state"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow/protobuf/go/flow/access"
+	gethCommon "github.com/onflow/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func Test_E2E_Previewnet_RemoteLedger(t *testing.T) {
+	ledger, err := newPreviewnetLedger()
+	require.NoError(t, err)
+
+	// this is a pre-established test account on previewnet
+	addrBytes, err := hex.DecodeString("BC9985a24c0846cbEdd6249868020A84Df83Ea85")
+	require.NoError(t, err)
+	testAddress := types.NewAddressFromBytes(addrBytes).ToCommon()
+
+	stateDB, err := state.NewStateDB(ledger, previewnetStorage)
+	require.NoError(t, err)
+
+	assert.NotEmpty(t, stateDB.GetCode(testAddress))
```

Comment:
If this `testAddress` is a smart contract account, can we maybe add more specific assertions? Instead of `assert.NotEmpty`. Just to make sure that we get back decent values.

Reply:
I don't want to copy paste complete contract code, this test tbh should be improved to deploy the contract and then check, but we don't have that much time now and this will all be replaced soon anyway with local state index.

Reply:
Sounds fair :+1: